purchase_triple_discount/tests/test_purchase_discount.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+class TestPurchaseOrder(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseOrder, cls).setUpClass()
+        cls.partner = cls.env['res.partner'].create({
+            'name': 'Mr. Odoo',
+        })
+        cls.product1 = cls.env['product.product'].create({
```

Comment:
here `cls.env['product.product']` use multiple time, so you should use as a global variable.